internal/datatransfer/monitor.go

```diff
@@ -112,56 +123,125 @@ func (m *Monitor) getPathsOfWorkload(workloadName string) []*models.DataPath {
 	return dataPaths
 }
 
+func (m *Monitor) ForceSync() error {
+	return m.syncPaths()
+}
+
+func (m *Monitor) HasStorageDefined() bool {
+	storage := m.config.GetDeviceConfiguration().Storage
+	if storage == nil {
+		return false
+	}
+
+	return storage.S3 != nil
+}
+
+func (m *Monitor) SetStorage(fs FileSync) {
+	m.syncMutex.Lock()
+	m.fsSync = fs
+	m.syncMutex.Unlock()
```

Comment:
nit: i like the pattern of `defer m.syncMutex.Unlock()`  right after we acquire the lock.
algought this is a short and clear function, but it will be consistent with the rest of the code and will reduce changes of surprises in the future if this function will be changed